concepts/default-enarx-trust-process.md

```diff
@@ -0,0 +1,480 @@
+now the **Empty Keep**.  This may have been via set-up information from
+the **Orchestrator**, as part of the communications with the **Enarx host
+agent**, or via "call-back" from the **Empty Keep**: in the latter case,
+the **Enarx client agent** must be certain (through cryptographic
+authentication) that the **Empty Keep** is the expected entity.  This may
+be achieved by the injection of authentication material into the **Empty
+Keep** at provision-time, but this must then be tied to the attestation
+measurement, and unique to each **Empty Keep** instance.  The session key
+established in the state 1->2 transition may be sufficient for this
+purpose.
+
+The **Enarx client agent** now creates an encrypted communications channel
+to the **Empty Keep**, using the session key established in the state 1->2
+transition.  The **Empty Keep** will be listening for a network connection,
+and once the encrypted communications channel is set up, the **Enarx client
+agent** transmits the **Tenant workload image** to the **Empty Keep**.  The
+**Empty Keep** receives the **Tenant workload image**, and instantiates it
+as a full workload.  This completes the transition of the **Empty Keep** to
+a **Provisioned Keep**.
+
+
+## Requirements
+### General requirements
+TODO: these requirements seem general to the project, and probably
+deserve their own RFC.  
+
+The security of the tenant (workload owner) MUST always be considered
+paramount when considering designs and implementations of Enarx
+components, protocols or other artifacts.
+
+When a trade-off between the security of the tenant and workload needs
+to be made against the security of any other party (including host owner),
+the security of the tenant MUST always win.
+
+When measures for a party other than the tenant are being considered,
+the general principle should be to choose the highest levels of security,
+but any choices MUST NOT compromise tenant security. 
+
+The Tenant Workload MUST never be unencrypted on the network (in transit)
+within the Enarx-managed process.
+
+The Tenant Workload MUST never be unencrypted in storage once it has been
+transmitted by the Enarx client agent.
+
+The Tenant Workload MUST never be unencrypted in RAM once it has been
+transmitted by the Enarx client agent.
+
+The Tenant Workload MUST never be available in unencrypted form on the host.
+
+### Other requirements
+The channel between the Enarx client agent and the Enarx host agent MAY be
+encrypted.
+
+The Enarx client agent SHOULD authenticate to the Enarx host agent.
+
+Integrity protection of the request SHOULD be applied.
+
+The protocol handshake and further steps SHOULD be encrypted.
+
+If integrity protection is to the protocol handshake and further
+communications between the Enarx client agent and the Enarx host agent
+are applied, all detected integrity failures SHOULD be reported to
+the Enarx client agent via the Enarx host agent where possible.
```

Comment:
Unless this is a grammatical turn of phrase i'm unfamiliar with, it seems simpler to me to say:
```suggestion
If integrity protection is applied to the protocol handshake and to further
communications between the Enarx client agent and the Enarx host agent,
all detected integrity failures SHOULD be reported to the Enarx client agent 
via the Enarx host agent where possible.
```

Reply:
Fixed.